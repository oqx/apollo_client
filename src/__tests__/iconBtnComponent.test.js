import React from "react";
import IconBtn from "../components/iconBtnComponent";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

describe("<IconBtn />", () => {
  let state = {
    btnClick: false
  };

  const spy = jest.fn();

  const handleClickStub = () => {
    state.btnClick = true;
    spy();
  };

  const iconBtn = shallow(
    <IconBtn iconClass="ion-ios-search" controlFunc={handleClickStub}>
      Submit
    </IconBtn>
  );

  it("should render correctly", () => {
    expect(toJson(iconBtn)).toMatchSnapshot();
  });

  it("should trigger a handler on click", () => {
    iconBtn.simulate("click");
    expect(spy).toHaveBeenCalledTimes(1);
    expect(state.btnClick).toBe(true);
  });
});
