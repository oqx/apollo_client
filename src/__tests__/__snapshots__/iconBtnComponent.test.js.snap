// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconBtn /> should render correctly 1`] = `
<div
  className="icon__default"
  onClick={[Function]}
  role="button"
>
  <i
    className="ion-ios-search"
  />
</div>
`;
